src/Infrastructure.VS.UnitTests/ActiveDocumentTrackerTests.cs

```diff
@@ -114,115 +116,85 @@ public void OnSelectionChanged_DoesNotRaiseEvent()
             var result = (testSubject as IVsSelectionEvents).OnSelectionChanged(null, 1, null, null, null, 1, null, null);
             result.Should().Be(VSConstants.S_OK);
 
-            mockEventHandler.VerifyNoOtherCalls();
-        }
-
-        [TestMethod]
-        public void OnElementValueChanged_NewValueIsNull_DoesNotRaiseEvent()
-        {
-            var result = SimulateElementValueChanged(VSConstants.VSSELELEMID.SEID_WindowFrame, null);
-            result.Should().Be(VSConstants.S_OK);
-
-            mockEventHandler.VerifyNoOtherCalls();
+            CheckEventNotRaised();
         }
 
         [TestMethod]
-        public void OnElementValueChanged_NonWindowFrameElement_DoesNotRaiseEvent()
+        [DataRow(VSConstants.VSSELELEMID.SEID_PropertyBrowserSID)]
+        [DataRow(VSConstants.VSSELELEMID.SEID_ResultList)]
+        [DataRow(VSConstants.VSSELELEMID.SEID_StartupProject)]
+        [DataRow(VSConstants.VSSELELEMID.SEID_UndoManager)]
+        [DataRow(VSConstants.VSSELELEMID.SEID_UserContext)]
+        [DataRow(VSConstants.VSSELELEMID.SEID_WindowFrame)]
+        public void OnElementValueChanged_NonDocumentElementId_DoesNotRaiseEvent(VSConstants.VSSELELEMID elementId)
         {
-            var selectedFrame = CreateMockFrame(__WindowFrameTypeFlags.WINDOWFRAMETYPE_Document);
+            var selectedFrame = Mock.Of<IVsWindowFrame>();
 
-            var result = SimulateElementValueChanged(VSConstants.VSSELELEMID.SEID_DocumentFrame, selectedFrame.Object);
-            result.Should().Be(VSConstants.S_OK);
-
-            mockEventHandler.VerifyNoOtherCalls();
-            selectedFrame.VerifyNoOtherCalls();
-        }
-
-        [TestMethod]
-        public void OnElementValueChanged_FrameElement_CantQueryFrameType_DoesNotRaiseEvent()
-        {
-            var selectedFrame = CreateFrameWithoutType();
-
-            var result = SimulateElementValueChanged(VSConstants.VSSELELEMID.SEID_WindowFrame, selectedFrame.Object);
+            var result = SimulateElementValueChanged(elementId, selectedFrame);
             result.Should().Be(VSConstants.S_OK);
 
             mockEventHandler.VerifyNoOtherCalls();
             textDocumentProviderMock.VerifyNoOtherCalls();
         }
 
         [TestMethod]
-        public void OnElementValueChanged_FrameElement_ToolFrame_DoesNotRaiseEvent()
+        public void OnElementValueChanged_DocumentElement_NewValueIsNull_RaisesEventWithNullArg()
```

Comment:
We should now raise the event with a null arg. It means the last doc has been closed, or the doc isn't a text doc (e.g. is a graphical doc).